### PR TITLE
userLeaderBoard and MdoLeaderBoard api changes

### PR DESCRIPTION
**Pull request by shankaragoudab**

### CODE REVIEW

File: src/main/java/org/sunbird/halloffame/service/HallOfFameServiceImpl.java [hunk 1]
```diff
@@ -146,6 +146,63 @@ public SBApiResponse fetchingTop10Learners(String ministryOrgId, String authToke
         return response;
     }
 
+    @Override
+    public SBApiResponse getUserLeaderBoard(String orgId) {
+        SBApiResponse response = ProjectUtil.createDefaultResponse(Constants.API_HALL_OF_FAME_ORG_READ);
+        if (orgId == null || orgId.isEmpty()) {
```
Comment (Sahil-tarento) [inline src/main/java/org/sunbird/halloffame/service/HallOfFameServiceImpl.java]: Please use StringUtils class

File: src/main/java/org/sunbird/halloffame/service/HallOfFameServiceImpl.java [hunk 2]
```diff
@@ -146,6 +146,63 @@ public SBApiResponse fetchingTop10Learners(String ministryOrgId, String authToke
...
+            setBadRequestResponse(response, Constants.INVALID_ORG_ID);
+            return response;
+        }
+        Map<String, Object> propertyMap = new HashMap<>();
+        propertyMap.put(Constants.ORGID, orgId);
+        propertyMap.put(Constants.DB_COLUMN_ROW_NUM, Arrays.asList(1, 2, 3, 4, 5, 6, 7, 8, 9, 10));
+        try {
+            List<Map<String, Object>> userLeaderBoard = cassandraOperation.getRecordsByPropertiesWithoutFiltering(
+                    Constants.KEYSPACE_SUNBIRD, Constants.NLW_USER_LEADERBOARD, propertyMap, null);
+            if (userLeaderBoard == null || userLeaderBoard.isEmpty()) {
```
Comment (Sahil-tarento) [inline src/main/java/org/sunbird/halloffame/service/HallOfFameServiceImpl.java]: Use CollectionUtils

File: src/main/java/org/sunbird/halloffame/service/HallOfFameServiceImpl.java [hunk 3]
```diff
@@ -146,6 +146,63 @@ public SBApiResponse fetchingTop10Learners(String ministryOrgId, String authToke
...
+    @Override
+    public SBApiResponse getUserLeaderBoard(String orgId) {
+        SBApiResponse response = ProjectUtil.createDefaultResponse(Constants.API_HALL_OF_FAME_ORG_READ);
+        if (orgId == null || orgId.isEmpty()) {
+            setBadRequestResponse(response, Constants.INVALID_ORG_ID);
+            return response;
+        }
+        Map<String, Object> propertyMap = new HashMap<>();
+        propertyMap.put(Constants.ORGID, orgId);
+        propertyMap.put(Constants.DB_COLUMN_ROW_NUM, Arrays.asList(1, 2, 3, 4, 5, 6, 7, 8, 9, 10));
```
Comment (Sahil-tarento) [inline src/main/java/org/sunbird/halloffame/service/HallOfFameServiceImpl.java:158]: Why this is set this way explain?

File: src/main/java/org/sunbird/halloffame/service/HallOfFameServiceImpl.java [hunk 4]
```diff
@@ -146,6 +146,63 @@ public SBApiResponse fetchingTop10Learners(String ministryOrgId, String authToke
...
+
+    @Override
+    public SBApiResponse getMdoLeaderBoard() {
+        SBApiResponse response = ProjectUtil.createDefaultResponse(Constants.API_HALL_OF_FAME_MDO_LEADERBOARD);
+        Map<String, Object> propertyMap = new HashMap<>();
+        propertyMap.put(Constants.SIZE, Arrays.asList("S", "M", "L", "XL"));
+        try {
+            List<Map<String, Object>> mdoLeaderBoard = cassandraOperation.getRecordsByPropertiesWithoutFiltering(
+                    Constants.KEYSPACE_SUNBIRD, Constants.NLW_MDO_LEADERBOARD, propertyMap, null);
+            if (mdoLeaderBoard == null || mdoLeaderBoard.isEmpty()) {
```
Comment (Sahil-tarento) [inline src/main/java/org/sunbird/halloffame/service/HallOfFameServiceImpl.java]: Use CollectionUtils

**Review by Sahil-tarento — CHANGES_REQUESTED** [review]

File: src/main/java/org/sunbird/halloffame/service/HallOfFameServiceImpl.java (hunk 3 above)
Comment (shankaragoudab) [reply src/main/java/org/sunbird/halloffame/service/HallOfFameServiceImpl.java:158]: this is to fetch only first 10 values from the table (specifically first for row_num column).
this is to show rank for the leader board, where first 10 only will be considered.

**Pull request merged**